R/write_sessioninfo.R

```diff
@@ -0,0 +1,71 @@
+#' Write essential session info to a file or connection
+#'
+#' Writes essentials of
+#' \code{\link[sessioninfo:session_info]{sessioninfo::session_info()}}
+#' to a file or connection in order to
+#' maintain versionable documentation of a workflow and aid
+#' reproducibility.
+#'
+#' The following information is written:
+#' \itemize{
+#' \item{essential platform info: version, os, system, ctype}
+#' \item{essential info on loaded packages: package, loadedversion, date,
+#' source if different from CRAN.}
+#' }
+#'
+#' @param file A file path or a \code{\link[base]{connection}}.
+#'
+#' @importFrom sessioninfo
+#' platform_info
+#' package_info
+#' @importFrom stringr
+#' str_detect
+#' @importFrom utils
+#' write.table
+#' capture.output
+#' @importFrom rlang
+#' .data
+#' @importFrom dplyr
+#' mutate
+#' select
+#'
+#' @export
+#'
+#' @examples
```

Comment:
I've added this example. @florisvdh How do you see the use of this function in a workflow? I have something like this in mind:

- calculate a hash on the created txt file with session info
- check the hash to make sure nothing changed

?

Reply:
The main purpose is to document the environment in which a versioned workflow (R script, Rmd) is being executed. Beside the pure documentation & traceability functionality, it eases troubleshooting when another user wants to reproduce the outcome (e.g. did I get an error because I use another package version?).

Versioning either the txt file or a hash file would make visible whether changes have occurred, but only versioning of the txt file will make clear which changes have occurred. The advantage of a hash is that it is smaller (that is the strategy used in git LFS), but I think this advantage is rather marginal here.

So the implementation (and my recommendation) goes like: use `write_sessioninfo()` at the end of each workflow in order to also document (and version) your environment, not just your code.

Reply:
I see, I made it too complex. That's OK.